### PR TITLE
Fix concerns with tck-dist

### LINKED ISSUE #737: [Bug]: Improve clean-room ability to run the Data TCK
### Specification Version

1.0.0-RC1

### Bug report

The current state of the Data TCK could use some improvement regarding the ability of a party creating a clean-room implementation of the spec to run the TCK against their impl. The current state of the data TCK requires a fair amount of a priori knowledge to even run the TCK.

- No info about how persistence provider is supposed to be run.
- No info about having CDI ready.
- Missing info or persistence.xml descriptor.
- Unclear if annotation provided needs to be run.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by KyleAure**
Fixes #737 

FYI @edburns @lukasj

Tested dependabot in my fork and it worked: 
Screenshot 2024-05-15 at 13 23 07

### CODE REVIEW

**Review by njr-11 — APPROVED** [review]

**Comment by KyleAure** [comment]
Wait to merge until we know if this needs to be included in 1.0.0 or if it can wait for 1.0.1

File: tck-dist/src/main/starter/ee-pom.xml [hunk 1]
```diff
@@ -118,12 +121,12 @@
   <dependency>
    <groupId>jakarta.servlet</groupId>
    <artifactId>jakarta.servlet-api</artifactId>
-   <version>6.0.0</version>
+   <version>6.0</version>
```
Comment (lukasj) [inline tck-dist/src/main/starter/ee-pom.xml:124]: this should have been `${jakarta.servlet.version}` instead I guess

File: tck-dist/src/main/starter/ee-pom.xml [hunk 2]
```diff
@@ -118,12 +121,12 @@
...
   </dependency>
   <dependency>
    <groupId>jakarta.enterprise</groupId>
    <artifactId>jakarta.enterprise.cdi-api</artifactId>
-   <version>4.0.1</version>
+   <version>4.0</version>
```
Comment (lukasj) [inline tck-dist/src/main/starter/ee-pom.xml:129]: this should have been `${jakarta.enterprise.cdi.version}` instead I guess

File: tck-dist/src/main/starter/se-pom.xml [hunk 3]
```diff
@@ -20,27 +20,27 @@
...
 
   <arquillian.version>1.7.0.Alpha13</arquillian.version>
   <junit.version>5.9.0</junit.version>
+  <shrinkwrap.version>1.2.6</shrinkwrap.version>
   <sigtest.version>2.3</sigtest.version>
-
+  
   <maven.dep.plugin.version>3.3.0</maven.dep.plugin.version>
   <maven.comp.plugin.version>3.10.1</maven.comp.plugin.version>
   <maven.surefire.plugin.version>2.22.2</maven.surefire.plugin.version>
```
Comment (lukasj) [inline tck-dist/src/main/starter/se-pom.xml:46]: 2.22.2 -> 3.2.5 (anything 3+ should be fine)

File: tck-dist/src/main/starter/ee-pom.xml [hunk 4]
```diff
@@ -33,9 +33,9 @@
   <maven.compiler.testRelease>17</maven.compiler.testRelease>
 
   <!-- Dependency and Plugin Versions -->
-  
-  <!-- Default - use local snapshot - can be overwritten by runner -->
-  <jakarta.data.version>1.0.0-SNAPSHOT</jakarta.data.version>
+  <jakarta.data.version>1.0.0</jakarta.data.version>
```
Comment (lukasj) [inline tck-dist/src/main/starter/ee-pom.xml:36]: note that the data-api version and data-tck version are two different things having different lifecycles. Accepting a TCK challenge will require new TCK version while keeping the API version untouched

File: tck-dist/src/main/starter/se-pom.xml [hunk 5]
```diff
@@ -99,28 +102,29 @@
...
+   <version>${jakarta.servlet.version}</version>
   </dependency>
+ <!-- APIs referenced by TCK that do require implementations for standalone tests -->
   <dependency>
    <groupId>jakarta.enterprise</groupId>
    <artifactId>jakarta.enterprise.cdi-api</artifactId>
-   <version>4.0.1</version>
+   <version>${jakarta.enterprise.cdi.version}</version>
   </dependency>
+  <!-- TODO add a CDI implementation that is supported by your Jakarta Data implementation -->
```
Comment (lukasj) [inline tck-dist/src/main/starter/se-pom.xml:127]: If having and supporting `@Inject` is what is wanted and expected, then having a dependency on `jakarta.inject:jakarta.inject-api` and saying _DI framework implementation_ probably expresses that better. CDI itself is just a DI framework, so it satisfies that...

**Pull request merged**